storage/src/test/java/org/apache/kafka/admin/RemoteTopicCrudTest.java

```diff
@@ -0,0 +1,692 @@
+
+            admin.createTopics(List.of(new NewTopic(testTopicName, numPartitions, numReplicationFactor)
+                .configs(topicConfig))).all().get();
+
+            var configs = new HashMap<ConfigResource, Collection<AlterConfigOp>>();
+            configs.put(new ConfigResource(ConfigResource.Type.TOPIC, testTopicName),
+                List.of(
+                    new AlterConfigOp(new ConfigEntry(TopicConfig.REMOTE_LOG_STORAGE_ENABLE_CONFIG, "false"),
+                        AlterConfigOp.OpType.SET),
+                    new AlterConfigOp(new ConfigEntry(TopicConfig.REMOTE_LOG_DELETE_ON_DISABLE_CONFIG, "true"),
+                        AlterConfigOp.OpType.SET)
+                ));
+            admin.incrementalAlterConfigs(configs).all().get();
+
+            var newProps = new HashMap<String, String>();
+            for (AlterConfigOp op : configs.get(new ConfigResource(ConfigResource.Type.TOPIC, testTopicName))) {
+                newProps.put(op.configEntry().name(), op.configEntry().value());
+            }
+
+            verifyRemoteLogTopicConfigs(newProps);
+        }
+    }
+
+    @ClusterTest(
+        serverProperties = {
+            @ClusterConfigProperty(key = RemoteLogManagerConfig.REMOTE_STORAGE_MANAGER_CLASS_NAME_PROP, value = "org.apache.kafka.admin.RemoteTopicCrudTest$MyRemoteStorageManager"),
+            @ClusterConfigProperty(key = RemoteLogManagerConfig.REMOTE_LOG_METADATA_MANAGER_CLASS_NAME_PROP, value = "org.apache.kafka.admin.RemoteTopicCrudTest$MyRemoteLogMetadataManager")
+        }
+    )
+    void testTopicDeletion() throws Exception {
+        try (var admin = cluster.admin()) {
+            MyRemoteStorageManager.DELETE_SEGMENT_EVENT_COUNTER.set(0);
+            var topicConfig = Map.of(
+                TopicConfig.REMOTE_LOG_STORAGE_ENABLE_CONFIG, "true",
+                TopicConfig.RETENTION_MS_CONFIG, "200",
+                TopicConfig.LOCAL_LOG_RETENTION_MS_CONFIG, "100"
+            );
+            admin.createTopics(List.of(new NewTopic(testTopicName, numPartitions, numReplicationFactor)
+                .configs(topicConfig))).all().get();
+
+            admin.deleteTopics(List.of(testTopicName)).all().get();
+
+            TestUtils.waitForCondition(() -> {
+                try {
+                    admin.describeTopics(List.of(testTopicName)).allTopicNames().get();
+                    return false;
+                } catch (Exception exception) {
+                    return exception.getCause() instanceof UnknownTopicOrPartitionException;
+                }
+            },  "Topic should be deleted");
+
+            TestUtils.waitForCondition(() ->
+                    numPartitions * MyRemoteLogMetadataManager.SEGMENT_COUNT_PER_PARTITION == MyRemoteStorageManager.DELETE_SEGMENT_EVENT_COUNTER.get(), 
+                "Remote log segments should be deleted only once by the leader");
+        }
+    }
+
+    private <T extends Throwable> T assertThrowsException(Class<T> expectedType,
```

Comment:
This could be replaced by `org.apache.kafka.test.TestUtils.assertFutureThrows`, right?